internal/pkg/cache/cache.go

```diff
@@ -162,9 +165,32 @@ func (h *Handle) GetEntry(cacheType string, hash string) (e *Entry, err error) {
 	return e, nil
 }
 
-func (h *Handle) CleanCache(cacheType string, dryRun bool, days int) (err error) {
-	dir := h.getCacheTypeDir(cacheType)
+func (h *Handle) CleanMultipleArchCache(cacheType string, dryRun bool, days int) (err error) {
+	parentDir := filepath.Dir(h.rootDir)
+	var archs []string
+	entries, err := os.ReadDir(parentDir)
+	if err != nil {
+		return err
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			archs = append(archs, entry.Name())
+		}
+	}
+
+	for _, arch := range archs {
+		dir := path.Join(parentDir, arch, cacheType)
+		err = h.cleanCache(dir, cacheType, dryRun, days)
+		if err != nil {
+			return
+		}
+	}
```

Comment:
Is it necessary (or helpful) to separately delete each subdirectory, instead of just deleting the whole directory?